File: src/main/java/kr/mjc/rachel/basicsyntax/WhenExpression.java
package kr.mjc.rachel.basicsyntax;

public class WhenExpression {
  public static String describe(Object obj) {
    if (obj.equals(1)) return "One";
    else if (obj.equals("Hello")) return "Greeting";
    else if (obj instanceof Long) return "Long";
    else if (!(obj instanceof String)) return "Not a string";
    else return "Unknown";
  }

  public static void main(String[] args) {
    System.out.println(describe(1));
    System.out.println(describe("Hello"));
    System.out.println(describe(1000L));
    System.out.println(describe(2));
    System.out.println(describe("other"));
  }
}

File: src/main/kotlin/kr/mjc/jacob/basicsyntax/WhenExpression.kt
/**
 * https://kotlinlang.org/docs/basic-syntax.html#when-expression
 */
package kr.mjc.jacob.basicsyntax

fun describe(obj: Any): String = when (obj) {
  1 -> "One"
  "Hello" -> "Greeting"
  is Long -> "Long"
  !is String -> "Not a string"
  else -> "Unknown"
}

fun main() {
  println(describe(1))
  println(describe("Hello"))
  println(describe(1000L))
  println(describe(2))
  println(describe("other"))
}